drivers/mac/controller.h

```diff
@@ -0,0 +1,20 @@
+#ifndef controller_h
+#define controller_h
+
+#import <GameController/GCController.h>
+
+@interface Controller : GCEventViewController
+@property NSString *ID;
+@property GCController *context;
+@property GCExtendedGamepad *profile;
++ (void) new:(NSDictionary *)in return:(NSString *)returnID;
++ (void) emitButton:(NSString *)ID Input:(int)input Value:(float)value Pressed:(BOOL)pressed;
++ (void) emitDpad:(NSString *)ID Input:(int)input X:(float)x Y:(float)y;
++ (void) close:(NSDictionary *)in return:(NSString *)returnID;
++ (void) listen:(NSDictionary *)in return:(NSString *)returnID;
+
+- (void) controllerConnected;
```

Comment:
No need to prefix by controller here since it is  already in a Controller object.
```objc
- (void ) connected;
- (void ) disconnected;
```

Reply:
changed 👍